test/support/generator/get-facility-generator.ts

```diff
@@ -0,0 +1,192 @@
+import { PROPERTIES } from '@ukef/constants';
+import { AcbsGetFacilityResponseDto } from '@ukef/modules/acbs/dto/acbs-get-facility-response.dto';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { GetFacilityByIdentifierResponseDto } from '@ukef/modules/facility/dto/get-facility-by-identifier-response.dto';
+import { Facility } from '@ukef/modules/facility/facility.interface';
+
+import { AbstractGenerator } from './abstract-generator';
+import { RandomValueGenerator } from './random-value-generator';
+
+export class GetFacilityGenerator extends AbstractGenerator<AcbsGetFacilityResponseDto, GenerateResult, GenerateOptions> {
+  constructor(protected readonly valueGenerator: RandomValueGenerator, protected readonly dateStringTransformations: DateStringTransformations) {
+    super(valueGenerator);
+  }
+
+  protected generateValues(): AcbsGetFacilityResponseDto {
+    return {
+      FacilityIdentifier: this.valueGenerator.string(),
```

Comment:
1. We could use valueGenerator.ukefId(), it would give correct and type, so no need to check length.
2. `stringOfNumericCharacters` and `string` both would work, but we could add check later. In NodeJS MDM API *api-tests* we check schema of GET responses. We could add schema check for real response, not just tests.
3. Exchange field types are working at the moment, so we should keep same types. 
  - Most are strings.
  - fields ending with "Code" are strings. Example creditRatingCode with value "14". In the past we had values like " ", "   ", "BLANK". Code fields are reference fields so they should be strings.
  - exposurePeriod value is number of months, it could be number in frontend, but in ACBS it is saved as ExternalReferenceIdentifier string
  - numeric fields are percentage, value and similar. For example probabilityOfDefault is numeric, percentage field.
  - ACBS documentation can help - https://uat-api-uk.acbsasp.com/Help/Api/GET-Portfolio-portfolioIdentifier-Facility-facilityIdentifier

Reply:
Cool, thanks Audrius. I've changed `facilityIdentifier` to use `valueGenerator.ukefId()` everywhere and otherwise kept it the same.